File: src/examples/gpio.py
#!/usr/bin/env python

from mpsse import *
from time import sleep

# Open mpsse in GPIO mode

io = MPSSE()
io.Open(0x403,0x6010, GPIO, interface=IFACE_B)

pin = GPIOH7
# Toggle the first GPIO pin on/off 10 times
for i in range(0, 10):
	print "set direction BCBUS7,6 output, other input", io.SetDirectionHigh(0xC0)
	print "set direction BDBUS7,6 output, other input", io.SetDirection(0xC0)
	io.PinHigh(pin)
	print "GPIO State:", io.PinState(pin)
	sleep(0.2)

	io.PinLow(pin)
	print "GPIO State:", io.PinState(pin)
	sleep(0.2)

print bin(io.ReadPins())
print bin(io.ReadPinsHigh())

io.Close()

File: src/examples/gpio.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <mpsse.h>

int main(void)
{
	struct mpsse_context *io = NULL;
	int i = 0, retval = EXIT_FAILURE;
	int j;

	io = MPSSE(GPIO, 1, 0);
	
	if(io && io->open)
	{
		for(i=0; i<10; i++)
		{
			for (j=GPIOL0; j <= GPIOL3; j++)
			{
				PinHigh(io, j);
				printf("GPIOL%d State: %d\n", j - GPIOL0, PinState(io, j, -1));
				usleep(100000);
			
				PinLow(io, j);
				printf("GPIOL%d State: %d\n", j - GPIOL0, PinState(io, j, -1));
			}
		}
	
		retval = EXIT_SUCCESS;
	}
	else
	{
		printf("Failed to open MPSSE: %s\n", ErrorString(io));
	}
		
	Close(io);

	return retval;
}
